### ISSUE TITLE
How to build Nix using Nix?

### ISSUE THREAD

**Issue by lukego** [NixOS/nix#1534]
I am building Nix from source for the first time. I would like to use another version of Nix to provide the build environment. Is this reasonable? How is it done? Do I need a particular version of nixpkgs?

Here are the errors I am seeing now with current master (94a0548d):

```
[luke@interlaken:~/git/nix]$ nix-build release.nix -A build
error: undefined variable ‘nlohmann_json’ at /home/luke/git/nix/release.nix:47:20
(use ‘--show-trace’ to show detailed location information)

[luke@interlaken:~/git/nix]$ nix-shell shell.nix
error: undefined variable ‘nlohmann_json’ at /home/luke/git/nix/shell.nix:21:7
(use ‘--show-trace’ to show detailed location information)
```

**Comment by edolstra** [comment]
Do you have an old version of Nixpkgs? `nlohmann_json` should be available in 17.03 and master.

BTW, it should be `-A build.x86_64-linux` (or some other platform), unless you want to build for all platforms at the same time.

**Comment by lukego** [comment]
It's not old by my standards, but it probably is by yours :).

Thanks for the hint. This command is working:

```
nix-build -I nixpkgs=http://github.com/nixos/nixpkgs/archive/release-17.03.tar.gz release.nix -A build.x86_64-linux
```
